demo/admin/src/products/ProductForm.tsx

```diff
@@ -150,54 +179,119 @@ export function ProductForm({ id }: FormProps): React.ReactElement {
+                                                            id
+                                                            used
+                                                        }
+                                                    }
+                                                }
+                                            `,
+                                        });
+
+                                        return data.manufacturerCountries.nodes;
+                                    }}
+                                    getOptionLabel={(option: GQLManufacturerCountriesQuery["manufacturerCountries"]["nodes"][0]) => option.id}
+                                    label="Manufacturer Country"
+                                    fullWidth
+                                />
+
+                                <AsyncSelectField
+                                    name="manufacturer"
+                                    loadOptions={async () => {
+                                        const { data } = await client.query<GQLManufacturersQuery, GQLManufacturersQueryVariables>({
+                                            query: gql`
+                                                query Manufacturers($filter: ManufacturerFilter) {
+                                                    manufacturers(filter: $filter) {
+                                                        nodes {
+                                                            id
+                                                            name
+                                                        }
+                                                    }
+                                                }
+                                            `,
+                                            variables: {
+                                                filter: {
+                                                    addressAsEmbeddable_country: {
+                                                        equal: values.manufacturerFilter?.id,
+                                                    },
+                                                },
+                                            },
+                                        });
+
+                                        return data.manufacturers.nodes;
+                                    }}
+                                    getOptionLabel={(option: GQLManufacturersQuery["manufacturers"]["nodes"][0]) => option.name}
+                                    label="Manufacturer"
+                                    fullWidth
+                                    disabled={!values?.manufacturerFilter}
+                                />
+
+                                <OnChangeField name="manufacturerFilter">
+                                    {(value, previousValue) => {
+                                        if (!isEqual(value, previousValue)) {
```

Comment:
yes, because else submitting mutation does change manufacturer resulting in an empty manufacturerFilter-field.

Reply:
Hmmm, strange 🤔 Could be a bug in the form's implementation... Should we pair on that?

Reply:
If you like to? btw I changed it to simply compare the id's because I think it's possible that options-query returns different fields than initial-values. (e.g. need to fetch some group to display options grouped, this wont be part of initial-values query) maybe this is the reason for the on-change-call...

Reply:
> btw I changed it to simply compare the id's because I think it's possible that options-query returns different fields than initial-values.

Yes, good idea! I was just concerned about the deep equality check.